### PR TITLE
Close builders that were created but not closed

### PR DESCRIPTION
**Pull request by bjhargrave**
Replaces and closes https://github.com/bndtools/bnd/pull/2090.

This includes the commit from 2090.

### CODE REVIEW

File: biz.aQute.bnd/src/aQute/bnd/main/bnd.java [hunk 1]
```diff
@@ -1779,11 +1782,14 @@ public void _debug(debugOptions options) throws Exception {
...
 			report(justif, "Project", project);
 
-			if (project.getSubBuilders() != null)
-				for (Builder sub : project.getSubBuilders()) {
-					report(justif, "Sub-Builder", sub);
-					getInfo(sub);
-				}
+			try (ProjectBuilder pb = project.getBuilder(null)) {
+				List<Builder> builders = pb.getSubBuilders();
+				if (builders != null)
```
Comment (seanbright) [inline biz.aQute.bnd/src/aQute/bnd/main/bnd.java]: Minor: Use braces for this `if` statement

File: biz.aQute.bndlib/src/aQute/bnd/build/Project.java [hunk 2]
```diff
@@ -2497,17 +2503,29 @@ public Builder getSubBuilder(File bndFile) throws Exception {
...
+		ProjectBuilder pb = getBuilder(null);
+		try {
+			for (Builder b : pb.getSubBuilders()) {
+				File propertiesFile = b.getPropertiesFile();
+				if (propertiesFile != null) {
+					if (propertiesFile.getCanonicalFile().equals(bndFile)) {
+						// Found it!
+						// disconnect from its parent life cycle
+						if (b == pb) {
+							pb = null;
```
Comment (seanbright) [inline biz.aQute.bndlib/src/aQute/bnd/build/Project.java]: Ewwww... I much prefer the `boolean close` method you use in `getSubBuilders()`

File: biz.aQute.bndlib/src/aQute/bnd/build/Project.java [hunk 3]
```diff
@@ -2517,12 +2535,25 @@ public Builder getSubBuilder(File bndFile) throws Exception {
...
-		for (Builder b : builders) {
-			if (b.getBsn().equals(string) || b.getBsn().endsWith("." + string))
-				return (ProjectBuilder) b;
+		ProjectBuilder pb = getBuilder(null);
+		try {
+			for (Builder b : pb.getSubBuilders()) {
+				if (b.getBsn().equals(string) || b.getBsn().endsWith("." + string)) {
+					// disconnect from its parent life cycle
+					if (b == pb) {
+						pb = null;
```
Comment (seanbright) [inline biz.aQute.bndlib/src/aQute/bnd/build/Project.java]: Same as above.

File: biz.aQute.bndlib/src/aQute/bnd/build/Project.java [hunk 4]
```diff
@@ -2729,13 +2800,15 @@ public Jar pack(String profile) throws Exception {
...
 
 	public void baseline() throws Exception {
-		ProjectBuilder b = getBuilder(null);
-		for (Builder pb : b.getSubBuilders()) {
-			ProjectBuilder ppb = (ProjectBuilder) pb;
-			Jar build = ppb.build();
-			getInfo(ppb);
+		try (ProjectBuilder parentBuilder = getBuilder(null)) {
+			for (Builder b : parentBuilder.getSubBuilders()) {
+				ProjectBuilder pb = (ProjectBuilder) b;
```
Comment (seanbright) [inline biz.aQute.bndlib/src/aQute/bnd/build/Project.java]: This reference is and cast are unnecessary. `Builder` has a `Jar build()` and `getInfo()` takes a `Reporter`

**Comment by bjhargrave** [comment]
@seanbright I addressed your comments and repushed.

**Pull request merged**